internal/buildscripts/packaging/tests/deployments/puppet/puppet_test.py

```diff
@@ -133,5 +138,78 @@ def test_puppet_without_fluentd(distro, puppet_release):
             verify_env_file(container)
             assert wait_for(lambda: service_is_running(container))
             assert container.exec_run("systemctl status td-agent").exit_code != 0
+            if distro in DEB_DISTROS:
+                assert container.exec_run("dpkg -s splunk-otel-auto-instrumentation").exit_code != 0
+            else:
+                assert container.exec_run("rpm -q splunk-otel-auto-instrumentation").exit_code != 0
         finally:
             run_container_cmd(container, f"journalctl -u {SERVICE_NAME} --no-pager")
+
+
+INSTRUMENTATION_CONFIG = string.Template(
+    f"""
+class {{ splunk_otel_collector:
+    splunk_access_token => '{SPLUNK_ACCESS_TOKEN}',
+    splunk_realm => '{SPLUNK_REALM}',
+    collector_version => '$version',
+    with_auto_instrumentation => true,
+    auto_instrumentation_version => '$version',
+    auto_instrumentation_resource_attributes => 'deployment.environment=test',
+    auto_instrumentation_service_name => 'test',
+}}
+"""
+)
+
+
+def verify_instrumentation_config(container):
+    config_path = "/usr/lib/splunk-instrumentation/instrumentation.conf"
+    libsplunk_path = "/usr/lib/splunk-instrumentation/libsplunk.so"
+    java_agent_path = "/usr/lib/splunk-instrumentation/splunk-otel-javaagent.jar"
+
+    try:
+        run_container_cmd(container, f"grep '^{libsplunk_path}' /etc/ld.so.preload")
+        run_container_cmd(container, f"grep '^java_agent_jar={java_agent_path}' {config_path}")
+        run_container_cmd(container, f"grep '^resource_attributes=deployment.environment=test' {config_path}")
+        run_container_cmd(container, f"grep '^service_name=test' {config_path}")
+    finally:
+        run_container_cmd(container, "cat /etc/ld.so.preload")
+        run_container_cmd(container, f"cat {config_path}")
+
+
+@pytest.mark.installer
+@pytest.mark.parametrize(
+    "distro",
+    [pytest.param(distro, marks=pytest.mark.deb) for distro in DEB_DISTROS]
+    + [pytest.param(distro, marks=pytest.mark.rpm) for distro in RPM_DISTROS],
+    )
+@pytest.mark.parametrize("puppet_release", PUPPET_RELEASE)
+def test_puppet_with_instrumentation(distro, puppet_release):
+    if "jessie" in distro and puppet_release != "6":
+        pytest.skip(f"Puppet release version {puppet_release} not supported on debian jessie")
+
+    if distro in DEB_DISTROS:
+        dockerfile = IMAGES_DIR / "deb" / f"Dockerfile.{distro}"
+    else:
+        dockerfile = IMAGES_DIR / "rpm" / f"Dockerfile.{distro}"
+
+    buildargs = {"PUPPET_RELEASE": puppet_release}
+    with run_distro_container(distro, dockerfile=dockerfile, path=REPO_DIR, buildargs=buildargs) as container:
+        try:
+            for version in ["0.48.0", "latest"]:
```

Comment:
Is there an easy way to confirm the desired version has been installed?

Reply:
Good point! The test has been updated for a basic version check.